packages/core/contracts/oracle/implementation/Staker.sol

```diff
@@ -129,10 +149,10 @@ abstract contract Staker is StakerInterface, Ownable, Lockable {
             voterStake.activeStake += amount;
             cumulativeActiveStake += amount;
         }
-
-        votingToken.transferFrom(msg.sender, address(this), amount);
+        if (from != address(this)) votingToken.transferFrom(from, address(this), amount);
```

Comment:
this line needs a comment. it's a bit nuanced as it enables some of the other magic you've got in this PR. consider explaining a bit more.